lib/esp_sdk/end_points/base.rb

```diff
@@ -78,8 +78,20 @@ def submit(url, type, options = {})
         @current_record  = ActiveSupport::HashWithIndifferentAccess.new(JSON.load(response.body))
       end
 
+      # Converts the link header into a hash of links
+      #
+      # "<http://test.host/api/v1/custom_signatures?page=2>; rel=\"last\", <http://test.host/api/v1/custom_signatures?page=2>; rel=\"next\""
+      # => { "last" => "http://test.host/api/v1/custom_signatures?page=2",
+      #      "next" => "http://test.host/api/v1/custom_signatures?page=2" }
       def pagination_links(response)
         @page_links = ActiveSupport::HashWithIndifferentAccess.new(JSON.load(response.headers[:link]))
+      rescue JSON::ParserError
+        @page_links = ActiveSupport::HashWithIndifferentAccess.new.tap do |page_links|
+          response.headers[:link].to_s.split(',').each  do |link|
+            /<(?<link>.*)>; rel="(?<key>\w*)"/ =~ link
+            page_links[key] = link
+          end
+        end
```

Comment:
Its a common best practice to only use rescues when the exceptions are exceptional. You shouldn't have a rescue when you are expecting an error. Rescue's are also less performant. That said, here I don't think there is a normal way to check for JSON without the exception. 

Related:
http://etdev.me/pragmatic/34/
http://technicaldebt.com/the-cost-of-using-rubys-rescue-as-logic/

Also, the headers needed to change. It is a public API so we need to follow standards/HTTP spec. A customer may use a library that would support the normal HTTP spec and shouldn't have to code around our overrides.